cpu/cc2538/startup-gcc.c

```diff
@@ -30,22 +30,23 @@
  *  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 /**
- * \addtogroup cc2538dk
+ * \addtogroup cc2538
  * @{
  *
  * \file
- * Startup code for the cc2538dk platform, to be used when building with gcc
+ * Startup code for the cc2538 chip, to be used when building with gcc
  */
 #include "contiki.h"
 #include "reg.h"
 #include "flash-cca.h"
 #include "sys-ctrl.h"
-#include "uart.h"
 
 #include <stdint.h>
 /*---------------------------------------------------------------------------*/
 extern int main(void);
 /*---------------------------------------------------------------------------*/
+#define WEAK_ALIAS(x) __attribute__ ((weak, alias(#x)))
```

Comment:
> SDCC, cc65

win32 too by the way

> How about this in `cc.h`?

I cannot see anything wrong with this, other than that traditionally we have not been checking for compiler family / version etc in `cc.h`, see for example how `CC_CONF_INLINE` is being used. This could also have been done automatically, but hasn't. I'm not sure about the original reasons though.

Would `__IAR_SYSTEMS_ICC__` not be sufficient?